### PR TITLE
Use API to get-or-create test student

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14182
affects: teacher
release note: Added Student View

Test plan:
* Install App Store version of Student app
* Install this build of Teacher app
* In teacher app, log in as narmstrong.instructure.com > t
* Use an existing test student:
  - Dashboard > CS 1400 > Student View > Should be logged in as Test Student
* In teacher app, log in as narmstrong.instructure.com > t2
* Create a new test student:
  - Dashboard > Modules course > Student View > Should be logged in as Test Student
* The Inbox tab should be hidden from test students

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/nlrcqc4) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fnlrcqc4)
teacher: [Link](https://rebrand.ly/8faarc1) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F8faarc1)
parent: [Link](https://rebrand.ly/akyzswr) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fakyzswr)

**Comment by inst-danger** [comment]
#### Release Note: 
Added Student View

#### Affected Apps: Teacher

[MBL-14182](https://instructure.atlassian.net/browse/MBL-14182)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.59% | 93.09% | 0.5%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: rn/Teacher/src/canvas-api/apis/users.js [hunk 1]
```diff
@@ -24,13 +24,8 @@ export function getCustomColors (): ApiPromise<CustomColors> {
...
 
-export function getFakeStudents (courseID) {
-  return httpClient.get(`courses/${courseID}/users`, {
-    params: {
-      include: ['test_student'],
-      enrollment_type: ['student_view'],
-    },
-  })
+export function getFakeStudent (courseID) {
+  return httpClient.get(`courses/${courseID}/student_view_student`)
```
Comment (vanwagonet) [inline rn/Teacher/src/canvas-api/apis/users.js:28]: nice. I didn't realize this would replace a call we already were making.

**Comment by vanwagonet** [comment]
I think I figured out the steps to repro. Hard kill the Student app before trying to use the Student View button. When the Student app was already running in the background it works correctly.

**Pull request merged**